tests/pos/i15398.scala

```diff
@@ -0,0 +1,9 @@
+object i15398 {
+  type Tuple23 = (Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int)
+
+  summon[Tuple.Size[Tuple23] =:= 23]
+  val m = summon[scala.deriving.Mirror.Of[Tuple23]]
+  summon[m.MirroredLabel =:= "Tuple23"]
```

Comment:
I guess this is the main bike-shedding moment - i.e. should it include arity, or just `Tuple` (or another variant)

Reply:
should also update the docs (or create an issue to) for Mirror generation here https://www.scala-lang.org/api/3.7.1/docs/docs/reference/contextual/derivation.html

Reply:
Created https://github.com/scala/scala3/issues/23475 for the docs because it was not immediately clear to me how to resolve it.

> case class can evolve over time to add new fields but its MirroredLabel doesn't change - so I dont think unique label is necessary.

For named tuples it always "NamedTuple". So based on that we should not include the arity, but should we then also drop it for arity < 22 to avoid inconsistency? 


> I guess this is the main bike-shedding moment - i.e. should it include arity, or just Tuple (or another variant) - i.e. 

Is there so good way to make progress on this. Feel like someone just need to make a decision.

Reply:
Well we should not change existing behaviour so I would just use Tuple as the label for size >=23

Reply:
Should I read that as you are suggesting changing the PR to have that behavior instead of including the arity? Or it just a comment that we should not break existing behavior.

Reply:
> but should we then also drop it for arity < 22 to avoid inconsistency?

no, because that would change the behavior. Anyway my suggestions above are what i meant :)